src/main/java/gov/usgs/earthquake/aws/AwsProductSender.java

```diff
@@ -116,7 +116,24 @@ public void sendProduct(final Product product) throws Exception {
       ) {
         LOGGER.fine("Getting upload urls for " + json.toString());
         // get upload urls, response is product with signed content urls for upload
-        Product uploadProduct = getUploadUrls(json);
+        Product uploadProduct;
+        try {
+          uploadProduct = getUploadUrls(json);
+        } catch (HttpException e) {
+          HttpURLConnection connection = e.response.connection;
+          // check for server error
+          if (connection.getResponseCode() >= 500) {
+            LOGGER.log(Level.FINE,
+                "[" + getName() + "] get upload urls exception, trying again", e);
+            // try again after random back off (1-5 s)
+            Thread.sleep(1000 + Math.round(4000 * Math.random()));
```

Comment:
This is probably fine, but AWS does recommend and exponential backoff. Maybe 10 * (10^retry) ms ? Not sure the retry number is tracked?

Reply:
Yeah, retries are tracked outside the listener...  Those retries jump to the end of the line, while this is an additional attempt inline.